beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -204,7 +266,12 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 		}
 	}
 
-	return s.sendBatchRootRequest(ctx, parentRoots, randGen)
+	// TODO(EIP-4844): Rework this to send blocks and sidecar requests in lock-step. It's a more robust way of handling partial failures
+	var err error
+	if err = s.sendBatchRootRequest(ctx, parentRoots, randGen); err == nil {
+		err = s.sendBatchSidecarRequest(ctx, missingSidecarRefs, randGen)
```

Comment:
This should use the same peer nodes. Otherwise, two different with different sets of blocks and sidecars could return conflicting info.